### ISSUE TITLE
[BUG][tag-v1.4.1]DSP panic when arecord via DMIC

### ISSUE THREAD

**Issue by dengyangchao** [thesofproject/sof#2245]
**Describe the bug**
DSP panic when change DMIC arecord between HW:0,6 and HW:0,7 quickly.
Same issue #2176  has been fixed on master through PR #2223, should back port to release v1.4.1

Can execute a script test1.sh like below:
```
arecord -Dhw:0,6 -fs16_le -c4 -r 48000 test.wav -vv -i -d 1
arecord -Dhw:0,7 -fs16_le -c4 -r 48000 test.wav -vv -i -d 1
arecord -Dhw:0,6 -fs16_le -c4 -r 48000 test.wav -vv -i -d 1
arecord -Dhw:0,7 -fs16_le -c4 -r 48000 test.wav -vv -i -d 1
```
Arecord -l list
```
**** List of CAPTURE Hardware Devices ****
card 0: sofhdadsp [sof-hda-dsp], device 0: HDA Analog (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofhdadsp [sof-hda-dsp], device 1: HDA Digital (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofhdadsp [sof-hda-dsp], device 6: DMIC48kHz (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofhdadsp [sof-hda-dsp], device 7: DMIC16kHz (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```
**To Reproduce**
1.Reboot system via"sudo reboot".
2.Wait runtime PM to suspend.
3.Execute script via./test1.sh
test1.sh.txt

**Reproduction Rate**
3/3

**Expected behavior**
Change dmic arecord successfully

Impact
Cause dsp panic

**Environment**
    * Kernel: https://github.com/thesofproject/linux/commits/topic/sof-dev 7fde4754
    * SOF: https://github.com/thesofproject/sof/commits/v1.4.1  fcf6c8c
    * Topology: file: topology/sof-hda-generic-4ch.tplg same as firmware
    * Platform: ICL-U RVP with onboard codec ALC700 in HDA mode

<details>
 <summary>Dmesg</summary>

````
[   77.397486] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x2024001e successful
[   77.397490] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60040000: GLB_STREAM_MSG: TRIG_START
[   77.397688] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60040000: GLB_STREAM_MSG: TRIG_START
[   77.397703] sof-audio-pci 0000:00:1f.3: error : DSP panic!
[   77.397712] sof-audio-pci 0000:00:1f.3: panic: dsp_oops_offset 788480 offset 788480
[   77.397720] sof-audio-pci 0000:00:1f.3: status: fw entered - code 00000005
[   77.397900] sof-audio-pci 0000:00:1f.3: error: runtime exception
[   77.397904] sof-audio-pci 0000:00:1f.3: error: trace point 00004000
[   77.397908] sof-audio-pci 0000:00:1f.3: error: panic at :0
[   77.397911] sof-audio-pci 0000:00:1f.3: error: DSP Firmware Oops
[   77.397916] sof-audio-pci 0000:00:1f.3: error: Exception Cause: LoadStorePIFDataErrorCause, Synchronous PIF data error during LoadStore access
[   77.397920] sof-audio-pci 0000:00:1f.3: EXCCAUSE 0x0000000d EXCVADDR 0x00000000 PS       0x00060325 SAR     0x00000000
[   77.397925] sof-audio-pci 0000:00:1f.3: EPC1     0xbe00da03 EPC2     0xbe01cee5 EPC3     0x00000000 EPC4    0x00000000
[   77.397929] sof-audio-pci 0000:00:1f.3: EPC5     0xbe00d0f6 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
[   77.397933] sof-audio-pci 0000:00:1f.3: EPS2     0x00060b20 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00060d20
[   77.397937] sof-audio-pci 0000:00:1f.3: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000222
[   77.397940] sof-audio-pci 0000:00:1f.3: stack dump from 0xbe04bfa0
[   77.397946] sof-audio-pci 0000:00:1f.3: 0xbe04bfa0: be00da03 00000000 00000000 00000000
[   77.397951] sof-audio-pci 0000:00:1f.3: 0xbe04bfa4: 00000000 00000000 be080c40 be080d00
[   77.397956] sof-audio-pci 0000:00:1f.3: 0xbe04bfa8: 4379d000 8b324fac 5b597028 ffffa383
[   77.397961] sof-audio-pci 0000:00:1f.3: 0xbe04bfac: 5b597028 ffffa383 00000099 00000000
[   77.397965] sof-audio-pci 0000:00:1f.3: 0xbe04bfb0: 59cb2000 ffffa383 5bb47140 ffffa383
[   77.397970] sof-audio-pci 0000:00:1f.3: 0xbe04bfb4: 5a9646c0 ffffa383 5c3b1100 ffffa383
[   77.397976] sof-audio-pci 0000:00:1f.3: 0xbe04bfb8: a3287d30 ffffffff 00000000 00000000
[   77.397981] sof-audio-pci 0000:00:1f.3: 0xbe04bfbc: 00000000 00000000 00000000 00000000
[   77.397984] sof-audio-pci 0000:00:1f.3: error: waking up any trace sleepers
[   77.905248] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 7 dir 1 cmd 0
[   77.905255] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[   78.408931] sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x60050000 size 12
[   78.408938] sof-audio-pci 0000:00:1f.3: info: preventing DSP entering D3 state to preserve context
[   78.408950] sof-audio-pci 0000:00:1f.3: status: fw entered - code 00000005
[   78.409065] sof-audio-pci 0000:00:1f.3: error: runtime exception
[   78.409068] sof-audio-pci 0000:00:1f.3: error: trace point 00004000
[   78.409070] sof-audio-pci 0000:00:1f.3: error: panic at :0
[   78.409073] sof-audio-pci 0000:00:1f.3: error: DSP Firmware Oops
[   78.409076] sof-audio-pci 0000:00:1f.3: error: Exception Cause: LoadStorePIFDataErrorCause, Synchronous PIF data error during LoadStore access
[   78.409080] sof-audio-pci 0000:00:1f.3: EXCCAUSE 0x0000000d EXCVADDR 0x00000000 PS       0x00060325 SAR     0x00000000
[   78.409083] sof-audio-pci 0000:00:1f.3: EPC1     0xbe00da03 EPC2     0xbe01cee5 EPC3     0x00000000 EPC4    0x00000000
[   78.409086] sof-audio-pci 0000:00:1f.3: EPC5     0xbe00d0f6 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
[   78.409089] sof-audio-pci 0000:00:1f.3: EPS2     0x00060b20 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00060d20
[   78.409092] sof-audio-pci 0000:00:1f.3: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000222
[   78.409095] sof-audio-pci 0000:00:1f.3: stack dump from 0xbe04bfa0
[   78.409100] sof-audio-pci 0000:00:1f.3: 0xbe04bfa0: be00da03 00000000 00000000 00000000
[   78.409103] sof-audio-pci 0000:00:1f.3: 0xbe04bfa4: 00000000 00000000 be080c40 be080d00
[   78.409107] sof-audio-pci 0000:00:1f.3: 0xbe04bfa8: 3f252e00 710a0e69 5b597028 ffffa383
[   78.409110] sof-audio-pci 0000:00:1f.3: 0xbe04bfac: 412afcdc ffffb626 00000000 00000000
[   78.409113] sof-audio-pci 0000:00:1f.3: 0xbe04bfb0: 5c759568 ffffa383 5c759570 ffffa383
[   78.409117] sof-audio-pci 0000:00:1f.3: 0xbe04bfb4: 5c187a00 ffffa383 5b597028 ffffa383
[   78.409120] sof-audio-pci 0000:00:1f.3: 0xbe04bfb8: 00000000 00000000 00000000 00000000
[   78.409123] sof-audio-pci 0000:00:1f.3: 0xbe04bfbc: 5c72f628 ffffa383 50130828 ffffa383
[   78.409137] sof-audio-pci 0000:00:1f.3: error: hda irq intsts 0x00000000 intlctl 0xc0000003 rirb 00
[   78.409139] sof-audio-pci 0000:00:1f.3: error: dsp irq ppsts 0x00000000 adspis 0x00000000
[   78.409160] sof-audio-pci 0000:00:1f.3: error: host status 0x00000000 dsp status 0x00000000 mask 0x00000003
[   78.409163] sof-audio-pci 0000:00:1f.3: error: waking up any trace sleepers
[   78.409718] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x20240000 successful
[   78.409732]  DMIC16kHz: ASoC: trigger FE cmd: 0 failed: -110
[   78.409775] sof-audio-pci 0000:00:1f.3: pcm: free stream 7 dir 1
[   78.409781] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[   78.912895] sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x60030000 size 12
[   78.912902] sof-audio-pci 0000:00:1f.3: info: preventing DSP entering D3 state to preserve context
[   78.912909] sof-audio-pci 0000:00:1f.3: status: fw entered - code 00000005
[   78.913024] sof-audio-pci 0000:00:1f.3: error: runtime exception
[   78.913027] sof-audio-pci 0000:00:1f.3: error: trace point 00004000
[   78.913029] sof-audio-pci 0000:00:1f.3: error: panic at :0
[   78.913032] sof-audio-pci 0000:00:1f.3: error: DSP Firmware Oops
[   78.913035] sof-audio-pci 0000:00:1f.3: error: Exception Cause: LoadStorePIFDataErrorCause, Synchronous PIF data error during LoadStore access
[   78.913039] sof-audio-pci 0000:00:1f.3: EXCCAUSE 0x0000000d EXCVADDR 0x00000000 PS       0x00060325 SAR     0x00000000
[   78.913042] sof-audio-pci 0000:00:1f.3: EPC1     0xbe00da03 EPC2     0xbe01cee5 EPC3     0x00000000 EPC4    0x00000000
[   78.913055] sof-audio-pci 0000:00:1f.3: EPC5     0xbe00d0f6 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
[   78.913058] sof-audio-pci 0000:00:1f.3: EPS2     0x00060b20 EPS3     0x00000000 EPS4     0x00000000 EPS5   
````
</details>

<details>
 <summary>logger-error-trace</summary>

````
CORE  LEVEL      COMP_ID                TIMESTAMP            DELTA                FILE_NAME	CONTENT
    0      2  SCHEDULE_LL         151211282.656250 151211280.000000 single_chan_domain.c:102 	dma_domain_notify_change()
    0      2          IPC         153258604.218750   2047321.500000    src/ipc/handler.c:459 	ipc: comp 60 -> trigger cmd 0x50000
    0      2         PIPE 10.67   153258637.291667        33.072918 src/audio/pipeline.c:664 	pipeline_trigger()
    0      2         HOST 10.60   153258644.583333         7.291667     src/audio/host.c:295 	host_trigger()
    0      2          DMA         153258653.333333         8.750000 intel/cavs/hda-dma.c:532 	hda-dmac: 4 channel 1 -> stop
    0      2       VOLUME 10.61   153258662.760417         9.427083 udio/volume/volume.c:548 	volume_trigger()
    0      2       EQ_IIR 10.62   153258670.781250         8.020833 udio/eq_iir/eq_iir.c:758 	eq_iir_trigger()
    0      2          DAI 10.66   153258678.958333         8.177083      src/audio/dai.c:526 	dai_comp_trigger(), command = 0
    0      2          DAI 10.66   153258686.302083         7.343750      src/audio/dai.c:585 	dai_comp_trigger(), PAUSE/STOP
    0      2          DMA         153258694.062500         7.760417 src/drivers/dw/dma.c:362 	dw_dma_stop(): dma 0 channel 0 stop
    0      2         DMIC         153258703.541667         9.479167 rs/intel/cavs/dmic.c:1429	dmic_trigger()
    0      2         DMIC         153258710.729167         7.187500 rs/intel/cavs/dmic.c:1353	dmic_stop()
    0      2         DMIC         153258718.802083         8.072917 rs/intel/cavs/dmic.c:1379	dmic_stop(), dmic_active_fifos = 1
    0      2  SCHEDULE_LL         153258739.739583        20.937500 single_chan_domain.c:354 	dma_single_chan_domain_unregister()
    0      2  SCHEDULE_LL         153258747.812500         8.072917 single_chan_domain.c:299 	dma_domain_unregister_owner()
    0      2  SCHEDULE_LL         153258755.000000         7.187500 single_chan_domain.c:156 	dma_single_chan_domain_irq_unregister()
    0      2          IPC         153259989.270833      1234.270874    src/ipc/handler.c:348 	ipc: comp 60 -> free
    0      2         PIPE 10.67   153260020.885417        31.614584 src/audio/pipeline.c:739 	pipeline_reset()
    0      2         HOST 10.60   153260028.177083         7.291667     src/audio/host.c:645 	host_reset()
    0      2       BUFFER         153260042.760417        14.583333   src/audio/buffer.c:110 	buffer_free()
    0      2       VOLUME 10.61   153260060.729167        17.968750 udio/volume/volume.c:690 	volume_reset()
    0      2       EQ_IIR 10.62   153260068.854167         8.125000 udio/eq_iir/eq_iir.c:887 	eq_iir_reset()
    0      2          DAI 10.66   153260080.729167        11.875000      src/audio/dai.c:488 	dai_reset()
    0      2       BUFFER         153260091.406250        10.677083   src/audio/buffer.c:110 	buffer_free()
    0      2          IPC         153278364.010417     18272.603516    src/ipc/handler.c:228 	ipc: comp 68 -> params
    0      2         PIPE 11.75   153278399.843750        35.833332 src/audio/pipeline.c:338 	pipeline_params()
    0      2         HOST 11.68   153278407.864583         8.020833     src/audio/host.c:458 	host_params()
    0      2       BUFFER         153278416.250000         8.385417   src/audio/buffer.c:24  	buffer_alloc()
    0      2          DMA         153278442.916667        26.666666 intel/cavs/hda-dma.c:399 	hda-dmac: 4 channel 1 -> get
    0      2          DMA         153278452.187500         9.270833 intel/cavs/hda-dma.c:581 	hda-dmac: 4 channel 1 -> config
    0      2       VOLUME 11.69   153278464.322917        12.135417 udio/volume/volume.c:254 	volume_params()
    0      2       EQ_IIR 11.70   153278473.541667         9.218750 udio/eq_iir/eq_iir.c:619 	eq_iir_params()
    0      2          DAI 11.74   153278482.500000         8.958333      src/audio/dai.c:340 	dai_params()
    0      2       BUFFER         153278490.781250         8.281250   src/audio/buffer.c:24  	buffer_alloc()
    0      2          DAI 11.74   153278510.572917        19.791666      src/audio/dai.c:303 	dai_capture_params() src_dev = 1 stream_id = 0 src_width = 4 dest_width = 4
    0      2          DAI 11.74   153278517.812500         7.239583      src/audio/dai.c:310 	dai_capture_params() fifo 10108
    0      2         PIPE 11.75   153278531.770833        13.958333 src/audio/pipeline.c:445 	pipeline_prepare()
    0      2         HOST 11.68   153278539.427083         7.656250     src/audio/host.c:602 	host_prepare()
    0      2       VOLUME 11.69   153278550.104167        10.677083 udio/volume/volume.c:608 	volume_prepare()
    0      2       EQ_IIR 11.70   153278561.770833        11.666667 udio/eq_iir/eq_iir.c:813 	eq_iir_prepare()
    0      2       EQ_IIR 11.70   153278569.895833         8.125000 udio/eq_iir/eq_iir.c:850 	eq_iir_prepare(), source_format=2, sink_format=0
    0      2       EQ_IIR         153278581.406250        11.510417 udio/eq_iir/eq_iir.c:402 	eq_iir_init_coef(), response assign for 2 channels, 1 responses
    0      2       EQ_IIR         153278589.635417         8.229167 udio/eq_iir/eq_iir.c:471 	eq_iir_init_coef(), ch 0 is set to response 0
    0      2       EQ_IIR         153278596.562500         6.927083 udio/eq_iir/eq_iir.c:471 	eq_iir_init_coef(), ch 1 is set to response 0
    0      2       EQ_IIR         153278603.697917         7.135417 udio/eq_iir/eq_iir.c:471 	eq_iir_init_coef(), ch 2 is set to response 0
    0      2       EQ_IIR         153278610.572917         6.875000 udio/eq_iir/eq_iir.c:471 	eq_iir_init_coef(), ch 3 is set to response 0
    0      2       EQ_IIR 11.70   153278623.333333        12.760417 udio/eq_iir/eq_iir.c:864 	eq_iir_prepare(), IIR is configured.
    0      2          DAI 11.74   153278632.552083         9.218750      src/audio/dai.c:440 	dai_prepare()
    0      2          IPC         153281232.760417      2600.208252    src/ipc/handler.c:459 	ipc: comp 68 -> trigger cmd 0x40000
    0      2         PIPE 11.75   153281268.906250        36.145832 src/audio/pipeline.c:664 	pipeline_trigger()
    0      2         HOST 11.68   153281276.458333         7.552083     src/audio/host.c:295 	host_trigger()
    0      2          DMA         153281285.000000         8.541667 intel/cavs/hda-dma.c:456 	hda-dmac: 4 channel 1 -> start
    0      2          DMA         153281292.864583         7.864583 intel/cavs/hda-dma.c:321 	hda-dmac: 4 channel 1 -> enable
    0      2       VOLUME 11.69   153281358.489583        65.625000 udio/volume/volume.c:548 	volume_trigger()
    0      2       EQ_IIR 11.70   153281366.770833         8.281250 udio/eq_iir/eq_iir.c:758 	eq_iir_trigger()
    0      2          DAI 11.74   153281375.156250         8.385417      src/audio/dai.c:526 	dai_comp_trigger(), command = 1
    0      2          DAI 11.74   153281382.395833         7.239583      src/audio/dai.c:537 	dai_comp_trigger(), START
    0      2         DMIC         153281389.531250         7.135417 rs/intel/cavs/dmic.c:1429	dmic_trigger()
    0      2         DMIC         153281396.927083         7.395833 rs/intel/cavs/dmic.c:1229	dmic_start()
    0      2         DMIC         153281410.781250        13.854167 rs/intel/cavs/dmic.c:1247	dmic_start(), dmic->fifo_b
    0      2         DMIC         153281418.541667         7.760417 rs/intel/cavs/dmic.c:1274	dmic_start(), mic_a = 1, mic_b = 1, fir_a = 1, fir_b = 1
    0      2         DMIC         153281426.979167         8.437500 rs/intel/cavs/dmic.c:1274	dmic_start(), mic_a = 1, mic_b = 1, fir_a = 1, fir_b = 1
    0      2         DMIC         153281434.947917         7.968750 rs/intel/cavs/dmic.c:1331	dmic_start(), dmic_active_fifos = 1
    0      2  SCHEDULE_LL         153281445.052083        10.104167 edule/timer_domain.c:47  	timer_domain_register()
    0      2         DMIC         153281455.468750        10.416667 rs/intel/cavs/dmic.c:1344	dmic_start(), done


````
</details>

[dmesg.log](https://github.com/thesofproject/sof/files/4012001/dmesg.log)
[logger.log](https://github.com/thesofproject/sof/files/4012002/logger.log)
[logger-error.log](https://github.com/thesofproject/sof/files/4012003/logger-error.log)

**Comment by lgirdwood** [comment]
Relevent context
Kernel
```
[   76.360924] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[   76.361199] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[   76.361204] sof-audio-pci 0000:00:1f.3: pcm: stream dir 1, posn mailbox offset is 790528
[   76.361825] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 6 dir 1 cmd 1
[   76.362369] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x2024001e successful
[   76.362373] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60040000: GLB_STREAM_MSG: TRIG_START
[   76.362608] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60040000: GLB_STREAM_MSG: TRIG_START
[   77.184888] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[   77.184928] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[   77.386166] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 6 dir 1 cmd 0
[   77.386175] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[   77.386344] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[   77.386860] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x240000 successful
[   77.386871] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[   77.387026] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[   77.387052] sof-audio-pci 0000:00:1f.3: pcm: free stream 6 dir 1
[   77.387161] sof-audio-pci 0000:00:1f.3: pcm: close stream 6 dir 1
[   77.395582] sof-audio-pci 0000:00:1f.3: pcm: open stream 7 dir 1
[   77.395586] sof-audio-pci 0000:00:1f.3: period min 192 max 16384 bytes
[   77.395588] sof-audio-pci 0000:00:1f.3: period count 2 max 16
[   77.395590] sof-audio-pci 0000:00:1f.3: buffer max 65536 bytes
[   77.395959] sof-audio-pci 0000:00:1f.3: pcm: hw params stream 7 dir 1
[   77.395964] sof-audio-pci 0000:00:1f.3: generating page table for 00000000b249d770 size 0xf9c0 pages 16
[   77.395972] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x240000 successful
[   77.395994] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
[   77.396000] sof-audio-pci 0000:00:1f.3: period_bytes:0x3e40
[   77.396001] sof-audio-pci 0000:00:1f.3: periods:4
[   77.396030] sof-audio-pci 0000:00:1f.3: stream_tag 2
[   77.396041] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[   77.396306] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[   77.396309] sof-audio-pci 0000:00:1f.3: pcm: stream dir 1, posn mailbox offset is 790604
[   77.396945] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 7 dir 1 cmd 1
[   77.397486] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x2024001e successful
[   77.397490] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60040000: GLB_STREAM_MSG: TRIG_START
[   77.397688] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60040000: GLB_STREAM_MSG: TRIG_START
[   77.397703] sof-audio-pci 0000:00:1f.3: error : DSP panic!
[   77.397712] sof-audio-pci 0000:00:1f.3: panic: dsp_oops_offset 788480 offset 788480
[   77.397720] sof-audio-pci 0000:00:1f.3: status: fw entered - code 00000005
[   77.397900] sof-audio-pci 0000:00:1f.3: error: runtime exception
[   77.397904] sof-audio-pci 0000:00:1f.3: error: trace point 00004000
[   77.397908] sof-audio-pci 0000:00:1f.3: error: panic at :0
[   77.397911] sof-audio-pci 0000:00:1f.3: error: DSP Firmware Oops
[   77.397916] sof-audio-pci 0000:00:1f.3: error: Exception Cause: LoadStorePIFDataErrorCause, Synchronous PIF data error during LoadStore access
[   77.397920] sof-audio-pci 0000:00:1f.3: EXCCAUSE 0x0000000d EXCVADDR 0x00000000 PS       0x00060325 SAR     0x00000000
[   77.397925] sof-audio-pci 0000:00:1f.3: EPC1     0xbe00da03 EPC2     0xbe01cee5 EPC3     0x00000000 EPC4    0x00000000
[   77.397929] sof-audio-pci 0000:00:1f.3: EPC5     0xbe00d0f6 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
[   77.397933] sof-audio-pci 0000:00:1f.3: EPS2     0x00060b20 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00060d20
[   77.397937] sof-audio-pci 0000:00:1f.3: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000222
[   77.397940] sof-audio-pci 0000:00:1f.3: stack dump from 0xbe04bfa0
[   77.397946] sof-audio-pci 0000:00:1f.3: 0xbe04bfa0: be00da03 00000000 00000000 00000000
[   77.397951] sof-audio-pci 0000:00:1f.3: 0xbe04bfa4: 00000000 00000000 be080c40 be080d00
[   77.397956] sof-audio-pci 0000:00:1f.3: 0xbe04bfa8: 4379d000 8b324fac 5b597028 ffffa383
[   77.397961] sof-audio-pci 0000:00:1f.3: 0xbe04bfac: 5b597028 ffffa383 00000099 00000000
[   77.397965] sof-audio-pci 0000:00:1f.3: 0xbe04bfb0: 59cb2000 ffffa383 5bb47140 ffffa383
[   77.397970] sof-audio-pci 0000:00:1f.3: 0xbe04bfb4: 5a9646c0 ffffa383 5c3b1100 ffffa383
[   77.397976] sof-audio-pci 0000:00:1f.3: 0xbe04bfb8: a3287d30 ffffffff 00000000 00000000
[   77.397981] sof-audio-pci 0000:00:1f.3: 0xbe04bfbc: 00000000 00000000 00000000 00000000
[   77.397984] sof-audio-pci 0000:00:1f.3: error: waking up any trace sleepers
[   77.905248] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 7 dir 1 cmd 0
[   77.905255] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[   78.408931] sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x60050000 size 12
[   78.408938] sof-audio-pci 0000:00:1f.3: info: preventing DSP entering D3 state to preserve context
```
Firmware
```
    0      2         PIPE 11.75   153278399.843750        35.833332 src/audio/pipeline.c:338 	pipeline_params()
    0      2         HOST 11.68   153278407.864583         8.020833     src/audio/host.c:458 	host_params()
    0      2       BUFFER         153278416.250000         8.385417   src/audio/buffer.c:24  	buffer_alloc()
    0      2          DMA         153278442.916667        26.666666 intel/cavs/hda-dma.c:399 	hda-dmac: 4 channel 1 -> get
    0      2          DMA         153278452.187500         9.270833 intel/cavs/hda-dma.c:581 	hda-dmac: 4 channel 1 -> config
    0      2       VOLUME 11.69   153278464.322917        12.135417 udio/volume/volume.c:254 	volume_params()
    0      2       EQ_IIR 11.70   153278473.541667         9.218750 udio/eq_iir/eq_iir.c:619 	eq_iir_params()
    0      2          DAI 11.74   153278482.500000         8.958333      src/audio/dai.c:340 	dai_params()
    0      2       BUFFER         153278490.781250         8.281250   src/audio/buffer.c:24  	buffer_alloc()
    0      2          DAI 11.74   153278510.572917        19.791666      src/audio/dai.c:303 	dai_capture_params() src_dev = 1 stream_id = 0 src_width = 4 dest_width = 4
    0      2          DAI 11.74   153278517.812500         7.239583      src/audio/dai.c:310 	dai_capture_params() fifo 10108
    0      2         PIPE 11.75   153278531.770833        13.958333 src/audio/pipeline.c:445 	pipeline_prepare()
    0      2         HOST 11.68   153278539.427083         7.656250     src/audio/host.c:602 	host_prepare()
    0      2       VOLUME 11.69   153278550.104167        10.677083 udio/volume/volume.c:608 	volume_prepare()
    0      2       EQ_IIR 11.70   153278561.770833        11.666667 udio/eq_iir/eq_iir.c:813 	eq_iir_prepare()
    0      2       EQ_IIR 11.70   153278569.895833         8.125000 udio/eq_iir/eq_iir.c:850 	eq_iir_prepare(), source_format=2, sink_format=0
    0      2       EQ_IIR         153278581.406250        11.510417 udio/eq_iir/eq_iir.c:402 	eq_iir_init_coef(), response assign for 2 channels, 1 responses
    0      2       EQ_IIR         153278589.635417         8.229167 udio/eq_iir/eq_iir.c:471 	eq_iir_init_coef(), ch 0 is set to response 0
    0      2       EQ_IIR         153278596.562500         6.927083 udio/eq_iir/eq_iir.c:471 	eq_iir_init_coef(), ch 1 is set to response 0
    0      2       EQ_IIR         153278603.697917         7.135417 udio/eq_iir/eq_iir.c:471 	eq_iir_init_coef(), ch 2 is set to response 0
    0      2       EQ_IIR         153278610.572917         6.875000 udio/eq_iir/eq_iir.c:471 	eq_iir_init_coef(), ch 3 is set to response 0
    0      2       EQ_IIR 11.70   153278623.333333        12.760417 udio/eq_iir/eq_iir.c:864 	eq_iir_prepare(), IIR is configured.
    0      2          DAI 11.74   153278632.552083         9.218750      src/audio/dai.c:440 	dai_prepare()
    0      2          IPC         153281232.760417      2600.208252    src/ipc/handler.c:459 	ipc: comp 68 -> trigger cmd 0x40000
    0      2         PIPE 11.75   153281268.906250        36.145832 src/audio/pipeline.c:664 	pipeline_trigger()
    0      2         HOST 11.68   153281276.458333         7.552083     src/audio/host.c:295 	host_trigger()
    0      2          DMA         153281285.000000         8.541667 intel/cavs/hda-dma.c:456 	hda-dmac: 4 channel 1 -> start
    0      2          DMA         153281292.864583         7.864583 intel/cavs/hda-dma.c:321 	hda-dmac: 4 channel 1 -> enable
    0      2       VOLUME 11.69   153281358.489583        65.625000 udio/volume/volume.c:548 	volume_trigger()
    0      2       EQ_IIR 11.70   153281366.770833         8.281250 udio/eq_iir/eq_iir.c:758 	eq_iir_trigger()
    0      2          DAI 11.74   153281375.156250         8.385417      src/audio/dai.c:526 	dai_comp_trigger(), command = 1
    0      2          DAI 11.74   153281382.395833         7.239583      src/audio/dai.c:537 	dai_comp_trigger(), START
    0      2         DMIC         153281389.531250         7.135417 rs/intel/cavs/dmic.c:1429	dmic_trigger()
    0      2         DMIC         153281396.927083         7.395833 rs/intel/cavs/dmic.c:1229	dmic_start()
    0      2         DMIC         153281410.781250        13.854167 rs/intel/cavs/dmic.c:1247	dmic_start(), dmic->fifo_b
    0      2         DMIC         153281418.541667         7.760417 rs/intel/cavs/dmic.c:1274	dmic_start(), mic_a = 1, mic_b = 1, fir_a = 1, fir_b = 1
    0      2         DMIC         153281426.979167         8.437500 rs/intel/cavs/dmic.c:1274	dmic_start(), mic_a = 1, mic_b = 1, fir_a = 1, fir_b = 1
    0      2         DMIC         153281434.947917         7.968750 rs/intel/cavs/dmic.c:1331	dmic_start(), dmic_active_fifos = 1
    0      2  SCHEDULE_LL         153281445.052083        10.104167 edule/timer_domain.c:47  	timer_domain_register()
    0      2         DMIC         153281455.468750        10.416667 rs/intel/cavs/dmic.c:1344	dmic_start(), done
```
Firmware trace is not as uptodate as error log above.

**Comment by perexg** [comment]
It looks like a serious issue. When will be 1.4.2 released?

**Comment by plbossart** [comment]
> It looks like a serious issue. When will be 1.4.2 released?

It's annoying but not necessarily a crippling problem IMO.

The microphone users will typically always use the full-band 48kHz input, the 16kHz path is really intended for application such as hotwording/event detection. switching between the two cases is unlikely.

To be honest, I don't even know why this path would be exposed to users in a regular HDaudio+PulseAudio case and without additional non-public libraries in the firmware? this 16khz path could confuse most Linux users, it's mostly used by ChromeOS.

**Comment by lgirdwood** [comment]
@plbossart IIUC, it looks like the 48kHz path on DMIC is also causing the issue when switching between PCM6 and 7 ?

@xiulipan @fredoh9 @zrombel it seems like we have a small hole in the CI stress test coverage (i.e. alternating playback and capture devices). We need a test that stress tests alternating capture/playback to multiple PCM devices (like the test in this report).

**Comment by xiulipan** [comment]
@lgirdwood We have had a test (https://github.com/thesofproject/sof-test/blob/master/test-case/check-playback.sh with -r 3 option) for this situation after we find the issue (https://github.com/thesofproject/sof/issues/2176) in master branch and got that fixed (https://github.com/thesofproject/sof/pull/2223).

**Comment by lgirdwood** [comment]
@plbossart @xiulipan ok thanks, I now have #2223 in the 1.4-stable branch

**Comment by slawblauciak** [comment]
Can you please confirm whether this is fixed as of #2223 ?

**Comment by lgirdwood** [comment]
I guess @dengyangchao is still on vacation, #2223 has been reported by others to fix this.
